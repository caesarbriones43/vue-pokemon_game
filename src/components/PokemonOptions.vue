<template>
  <div class="options-container">
    <ul>
      <div v-if="isDisabled">
        <li
          class="disabled"
          v-for="pokemon in pokemons"
          :key="pokemon.id"
          @click="$emit('selection', pokemon.id)"
        >
          {{ pokemon.name }}
        </li>
      </div>
      <div v-else>
        <li
          v-for="pokemon in pokemons"
          :key="pokemon.id"
          @click="$emit('selection', pokemon.id)"
        >
          {{ pokemon.name }}
        </li>
      </div>
    </ul>
  </div>
</template>

<script>
export default {
  name: "PokemonOptions",
  props: {
    pokemons: {
      type: Array,
      required: true,
    },
    isDisabled: {
      type: Boolean,
      required: true,
    },
  },
};
</script>

<style scoped>
/* Pokemon Options */

.disabled {
  pointer-events: none;
  opacity: 0.6;
}

ul {
  list-style-type: none;
}
li {
  background-color: white;
  border-radius: 5px;
  border: 1px solid rgba(0, 0, 0, 0.2);
  cursor: pointer;
  margin-bottom: 10px;
  width: 250px;
}

li:hover {
  background-color: rgba(0, 0, 0, 0.05);
}

.options-container {
  display: flex;
  justify-content: center;
}
</style>
