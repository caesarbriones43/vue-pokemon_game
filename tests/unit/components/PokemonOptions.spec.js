import { shallowMount } from "@vue/test-utils";
import PokemonOptions from "@/components/PokemonOptions.vue";
import { mockPokemons } from "../mocks/pokemons.mocks";

describe("PokemonOptions Component", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(PokemonOptions, {
      props: {
        pokemons: mockPokemons,
        isDisabled: false,
      },
    });
  });

  test("should match whit snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("should show the 4 options correctly", () => {
    wrapper = shallowMount(PokemonOptions, {
      props: {
        pokemons: mockPokemons,
        isDisabled: false,
      },
    });

    const liTags = wrapper.findAll("li");

    expect(liTags.length).toBe(4);
    expect(liTags[0].text()).toBe("bulbasaur");
    expect(liTags[1].text()).toBe("ivysaur");
    expect(liTags[2].text()).toBe("venusaur");
    expect(liTags[3].text()).toBe("charmander");
  });

  test('should show emiter "selection" whit his parameters when click it', () => {
    const [l1, l2, l3, l4] = wrapper.findAll("li");

    l1.trigger("click");
    l2.trigger("click");
    l3.trigger("click");
    l4.trigger("click");

    expect(wrapper.emitted("selection").length).toBe(4);
    expect(wrapper.emitted("selection")[0]).toEqual([1]);
    expect(wrapper.emitted("selection")[1]).toEqual([2]);
    expect(wrapper.emitted("selection")[2]).toEqual([3]);
    expect(wrapper.emitted("selection")[3]).toEqual([4]);
  });
});
