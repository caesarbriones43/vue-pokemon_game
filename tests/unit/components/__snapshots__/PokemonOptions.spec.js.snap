// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonOptions Component should match whit snapshot 1`] = `
<div class="options-container">
  <ul>
    <div>
      <li>bulbasaur</li>
      <li>ivysaur</li>
      <li>venusaur</li>
      <li>charmander</li>
    </div>
  </ul>
</div>
`;
